index.js

```diff
@@ -28,7 +29,7 @@ module.exports = opts => {
 		try {
 			report = xo.lintText(file.contents.toString(), {
 				cwd: file.cwd,
-				filename: file.relative
+				filename: path.relative(file.cwd, file.path)
```

Comment:
Why is that your opinion? `file.basename` is the filename part of the path, which is exactly what `xo.lintText` expects.

https://github.com/gulpjs/vinyl#filebasename